### PR TITLE
Added functions to modify the data at an entry for list and slist. (Issue #11)

### PR DESCRIPTION
**Pull request by TheStefan**
See issue #11

### CODE REVIEW

File: src/list.c [hunk 1]
```diff
@@ -142,6 +142,14 @@ ListValue list_data(ListEntry *listentry)
 	return listentry->data;
 }
 
+void list_set(ListEntry *listentry, ListValue value)
+{
+	if (listentry != NULL) {
+		listentry->data = value;
+	}
+}
+
```
Comment (fragglet) [inline src/list.c]: Spurious double empty line.

File: src/list.h [hunk 2]
```diff
@@ -177,6 +179,14 @@ ListEntry *list_next(ListEntry *listentry);
 ListValue list_data(ListEntry *listentry);
 
 /**
+ * Set the value at a list entry. If listentry is NULL nothing is done.
+ *
+ * @param listentry 	Pointer to the list entry.
+ * @param value			The value to set.
+ */
+void list_set(ListEntry *listentry, ListValue value);
```
Comment (fragglet) [inline src/list.h]: Can we name it `list_set_data` to match `list_data`, and also make clear that 'set' is a verb and not referring to the 'set' data type?

**Comment by TheStefan** [comment]
I committed your advices!

**Comment by TheStefan** [comment]
Ping.

File: src/list.h [hunk 3]
```diff
@@ -46,6 +46,8 @@ CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  * To access an entry in the list by index, use @ref list_nth_entry or
  * @ref list_nth_data.
  *
+ * To modify data in the list use @ref list_set.
```
Comment (fragglet) [inline src/list.h]: Comment is out of date.

File: src/slist.h [hunk 4]
```diff
@@ -59,6 +59,7 @@ CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
  *
  * @li To find the next entry, use @ref slist_next.
  * @li To access the value stored at the entry, use @ref slist_data.
+ * @li To set the value stored at the entry, use @ref slist_set.
```
Comment (fragglet) [inline src/slist.h]: Comment is out of date.

**Comment by TheStefan** [comment]
Done

**Comment by TheStefan** [comment]
aw shit, please ignore this some branching stuff went wrong

**Comment by TheStefan** [comment]
It should be alright now.

**Comment by TheStefan** [comment]
Thnx :+1:

**Pull request merged**